modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -364,6 +368,8 @@ public CompletableFuture<Void> startAsync(ComponentContext componentContext) {
 
         rebalanceRetryDelayConfiguration.init();
 
+        placementDriver.listen(PrimaryReplicaEvent.PRIMARY_REPLICA_EXPIRED, this::onPrimaryReplicaExpired);
```

Comment:
Should we remove the catalog and place driver listeners when this manager stops?

Reply:
I guess it's not required. Neither there's such code in the source that was used to port given logic. I see that you've created a ticket for that https://issues.apache.org/jira/browse/IGNITE-24673 Let's discuss it there, WDYT?

Reply:
:ok_hand: